### PR TITLE
Added Qemu builder & compress postprocessors

### PR DESCRIPTION
**Pull request by biemond**
Hi 

Really great work.

I just added support for Qemu builder and Compress post processor

This way you can use this to build a image which can be used on openstack or some other cloud provider

Example

```
          os = "myqemu"
          pconfig = Packer::Config.new "#{os}.json"
          pconfig.description "#{os} xxx"

          pconfig.add_variable 'version', '1.0'

          builder = pconfig.add_builder Packer::Builder::QEMU
          builder.output_directory "#{os}-qemu"

          builder.accelerator 'kvm'
          builder.disk_interface 'virtio'
          builder.qemu_binary '/usr/libexec/qemu-kvm'
          builder.format 'raw'

          builder.http_directory 'http'
          builder.boot_command ["<tab> text ks=http://#{pconfig.macro.HTTPIP}:#{pconfig.macro.HTTPPort}/#{os}/ks.cfg<enter><wait>"]
          builder.boot_wait '12s'
          builder.disk_size 8096

          builder.headless true

          builder.iso_checksum 'xxx'
          builder.iso_checksum_type 'sha1'
          builder.iso_url '/tmp/some.iso'
          builder.shutdown_command "/sbin/halt -h -p"
          builder.communicator "ssh"
          builder.ssh_password "root"
          builder.ssh_port 22
          builder.ssh_username "root"
          builder.ssh_timeout "10000s"

          builder.net_device "virtio-net"
          builder.qemuargs [
            [
              "-m",
              "1024"
            ],
            [
              "-machine",
              "type=pc,accel=kvm"
            ],
            [
              "-device",
              "virtio-net-pci,netdev=user.0"
            ]
          ]

          builder.vm_name "#{os}"

          provisioner = pconfig.add_provisioner Packer::Provisioner::SHELL
          provisioner.scripts [
            'scripts/rhel_6_x_network.sh',
            'scripts/compact.sh'
          ]

          postprocessor = pconfig.add_postprocessor Packer::PostProcessor::COMPRESS
          postprocessor.output File.join('builds', pconfig.macro.Provider, "#{os}.tar.gz")

          pconfig.validate
          pconfig.build(quiet: false)
```

thanks

### CODE REVIEW

**Comment by ianchesal** [comment]
Hi and thank you for your contribution to the project. It looks like your code isn't passing the rubocop tests and your new build is missing a spec test for it. At a minimum it needs to pass the rubocop checks and ideally you'd also include a spec that ensures the new build's required options are enforced as expected.

For an example spec see: https://github.com/ianchesal/packer-config/blob/master/spec/packer/builders/vmware_vmx_spec.rb

Thanks!

**Comment by biemond** [comment]
I fixed the qemu rubocop error, the rest 4 issues is in code I didn't touch.
I will add a qemu rspec test

**Comment by biemond** [comment]
Added a basic qemu rspec test

**Comment by ianchesal** [comment]
This is [failing rubocop](https://travis-ci.org/ianchesal/packer-config/jobs/120593020) -- will need those cops passing. Thanks.

File: spec/packer/builders/qemu_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Encoding: utf-8
+require 'spec_helper'
+
+RSpec.describe Packer::Builder::Qemu do
+  let(:builder) { Packer::Builder.get_builder(Packer::Builder::QEMU) }
+  let(:in_commands_strings)  { [["command1", "1"], ["command2", "2"]] }
+  let(:in_commands_mixed)    { [["command1",  1 ], ["command2",  2 ]] }
+  let(:out_commands_strings) { [["command1", "1"], ["command2", "2"]] }
+
+  it 'has a type of virtualbox-iso' do
```
Comment (ianchesal) [inline spec/packer/builders/qemu_spec.rb]: Spec description is wrong. It's not looking for virtualbox-iso type.

File: spec/packer/builders/qemu_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Encoding: utf-8
+require 'spec_helper'
+
+RSpec.describe Packer::Builder::Qemu do
+  let(:builder) { Packer::Builder.get_builder(Packer::Builder::QEMU) }
+  let(:in_commands_strings)  { [["command1", "1"], ["command2", "2"]] }
```
Comment (ianchesal) [inline spec/packer/builders/qemu_spec.rb]: Never referenced in the specs. Can be removed.

**Comment by ianchesal** [comment]
If you rebase off master, those failing cops have already been fixed.

**Comment by ianchesal** [comment]
If you rebase off `develop` Rubocop checks are up to date now.

**Comment by ianchesal** [comment]
I'm _finally_ getting around to cleaning this up and merging it. I'll open a new pull request with your changes to get it merged. Will tag you in that PR. Closing this one.

**Pull request closed without merging**